### PR TITLE
Added new quest in Kanto for Bill's Grandpa

### PR DESCRIPTION
**Pull request by LuchoDreamer**
This quest is to help alleviate the QP wall from Kanto described in the #1071 approved suggestion.

This PR consists of several things:

Kanto map:

1. Adds Bill's House in Kanto (I was surprised that this place was never in the game to start with). It doesn't mess with Suicune's sprite during its quest. This house will be used for the whole quest.
2. Fixes a small cut of top of a tree that shouldn't be there next to Mt Moon. Mentioned by Vodo in discord.

Sprites:

1. Two Bill's Grandpa sprites made by courtesy of Drakay

Quest:

1. The quest consists, in its majority, of a pokémon treasure hunt based on both gen 2 and its remakes in gen 4, where if you present specific pokémon to Bill's Grandpa in the order that he asks them for, he will reward you each time with a evolution stone.
2. Majority of the quests consist of catching the specific pokémon (with a hint, not said outright), and defeating 100 pokémon of that same type to add a little bit of extra difficulty/time to it.
3. Most of these pokémon are the same he asks for during the game, one of them is actually a gen 2 pokémon, so I changed it for a gen 1 pokémon, and another one of them is changed because it's a QP pokémon and it would kinda defeat the purpose of the quest.
4. As said at the start, this quest is aimed to lessen the QP wall in Kanto, thus this whole quest has a value of 16500 QP, but it's not 16500 QP as reward, the rewards are evo items/pokémon with that value in total (keeping it a bit vague to not destroy the surprise, but it's probably easy enough to guess to be honest).
5. This quest also aims at keeping you busy for most of Kanto with it. You unlock it when you get access to Bill's House (after route 25, pretty early), and you can get its first two parts done pretty fast, the third part requires either after Koga or an egg (and as far as I know new players don't focus on eggs until much later on). And after two more parts, it ends with a temporary battle, that is balanced with Giovanni's gym in mind (Around 5% stronger than Giovanni).
6. Every step of the quest gives you one different reward.
7. After the quest is done, Bill's Grandpa remains in the house, house-sitting, so it's not empty.

This is all, if the rewards are too much, I can rebalance them, but I really think these rewards are fair since the quest keeps you occupied during most of the Kanto journey.

I hope the quest is liked o/.

### CODE REVIEW

File: src/scripts/quests/QuestLineHelper.ts [hunk 1]
```diff
@@ -135,6 +135,154 @@ class QuestLineHelper {
...
+
+        const pinkBalloon = new CaptureSpecificPokemonQuest('Jigglypuff', 'Catch the desired Pokémon', 1);
+        const punchNormal = new CustomQuest(100, 0, 'Defeat 100 Normal-type Pokémon.', () => {
+            return pokemonMap.filter(p => p.type.includes(PokemonType.Normal)).map(p => App.game.statistics.pokemonDefeated[p.id]()).reduce((a,b) => a + b, 0);
+        });
+
+        BillsGrandpaQuestLine.addQuest(new MultipleQuestsQuest([
+            pinkBalloon,
+            punchNormal,
+        ],'Bill\'s Grandpa want you to catch a Pokémon that is pink and like a balloon.'));
```
Comment (zyx-xyzzy) [inline src/scripts/quests/QuestLineHelper.ts:153]: want -> wants

File: src/scripts/quests/QuestLineHelper.ts [hunk 2]
```diff
@@ -135,6 +135,154 @@ class QuestLineHelper {
...
+
+        const blueRound = new CaptureSpecificPokemonQuest('Oddish', 'Catch the desired Pokémon', 1);
+        const punchGrass = new CustomQuest(100, 0, 'Defeat 100 Grass-type Pokémon.', () => {
+            return pokemonMap.filter(p => p.type.includes(PokemonType.Grass)).map(p => App.game.statistics.pokemonDefeated[p.id]()).reduce((a,b) => a + b, 0);
+        });
+
+        BillsGrandpaQuestLine.addQuest(new MultipleQuestsQuest([
+            blueRound,
+            punchGrass,
+        ],'Bill\'s Grandpa want you to catch a Pokémon that is round, blue and has has leaves growing on its head'));
```
Comment (zyx-xyzzy) [inline src/scripts/quests/QuestLineHelper.ts:177]: add comma after “blue”

File: src/scripts/quests/QuestLineHelper.ts [hunk 3]
```diff
@@ -135,6 +135,154 @@ class QuestLineHelper {
...
+
+        const redSphere = new CaptureSpecificPokemonQuest('Staryu', 'Catch the desired Pokémon', 1);
+        const punchWater = new CustomQuest(100, 0, 'Defeat 100 Water-type Pokémon.', () => {
+            return pokemonMap.filter(p => p.type.includes(PokemonType.Water)).map(p => App.game.statistics.pokemonDefeated[p.id]()).reduce((a,b) => a + b, 0);
+        });
+
+        BillsGrandpaQuestLine.addQuest(new MultipleQuestsQuest([
+            redSphere,
+            punchWater,
+        ],'Bill\'s Grandpa want you to catch a Pokémon that is has a red sphere in its body and is shaped like a star'));
```
Comment (zyx-xyzzy) [inline src/scripts/quests/QuestLineHelper.ts:201]: want -> wants
is has -> has

File: src/scripts/quests/QuestLineHelper.ts [hunk 4]
```diff
@@ -135,6 +135,154 @@ class QuestLineHelper {
...
+
+        const loyalRoar = new CaptureSpecificPokemonQuest('Growlithe', 'Catch the desired Pokémon', 1);
+        const punchFire = new CustomQuest(100, 0, 'Defeat 100 Fire-type Pokémon.', () => {
+            return pokemonMap.filter(p => p.type.includes(PokemonType.Fire)).map(p => App.game.statistics.pokemonDefeated[p.id]()).reduce((a,b) => a + b, 0);
+        });
+
+        BillsGrandpaQuestLine.addQuest(new MultipleQuestsQuest([
+            loyalRoar,
+            punchFire,
+        ],'Bill\'s Grandpa want you to catch a Pokémon that is very loyal and supposedly roars pretty well'));
```
Comment (zyx-xyzzy) [inline src/scripts/quests/QuestLineHelper.ts:225]: want -> wants

File: src/scripts/quests/QuestLineHelper.ts [hunk 5]
```diff
@@ -135,6 +135,154 @@ class QuestLineHelper {
...
+
+        const yellowAndRed = new CaptureSpecificPokemonQuest('Pikachu', 'Catch the desired Pokémon', 1);
+        const punchElectric = new CustomQuest(100, 0, 'Defeat 100 Electric-type Pokémon.', () => {
+            return pokemonMap.filter(p => p.type.includes(PokemonType.Electric)).map(p => App.game.statistics.pokemonDefeated[p.id]()).reduce((a,b) => a + b, 0);
+        });
+
+        BillsGrandpaQuestLine.addQuest(new MultipleQuestsQuest([
+            yellowAndRed,
+            punchElectric,
+        ],'Bill\'s Grandpa want you to catch a Pokémon that has a yellow body and red cheeks'));
```
Comment (zyx-xyzzy) [inline src/scripts/quests/QuestLineHelper.ts:249]: want -> wants

File: src/scripts/towns/TownList.ts [hunk 6]
```diff
@@ -321,6 +321,70 @@ const CeruleanSuperNerd = new NPC('Super Nerd Jovan', [
...
+    'Hm? You know Bill? He\'s my grandson. He\'s not here right now. He does something with PCs, so I\'m house-sitting.',
+    'I like Pokémon but I\'m not as knowledgeable as my grandson. My grandson Bill told me about a Pokémon that is pink and like a balloon.',
+    'Would you catch that Pokémon and show it to me, please?',
+], {image: 'assets/images/npcs/Bill\'s Grandpa with Eevee.png',
+    requirement: new MultiRequirement([new QuestLineStartedRequirement('Bill\'s Grandpa Treasure Hunt'), new QuestLineStepCompletedRequirement('Bill\'s Grandpa Treasure Hunt', 1, GameConstants.AchievementOption.less)]),
+});
+
+const BillsGrandpa2 = new NPC('Bill\'s Grandpa', [
+    'Ah, so that is Jigglypuff? Isn\'t it cute! That\'s so kind of you. Thanks! This Moon Stone is a token of my appreciation.',
+    'Ah, my grandson mentioned a round, green Pokémon, wait not green, it\'s blue, a blue Pokémon that has leaves growing on its head',
```
Comment (zyx-xyzzy) [inline src/scripts/towns/TownList.ts:334]: Missing punctuation at the end of the sentence
(note: this also applies to several of the quest-step descriptions, but idk if those are intentional or not…)

File: src/scripts/towns/TownList.ts [hunk 7]
```diff
@@ -321,6 +321,70 @@ const CeruleanSuperNerd = new NPC('Super Nerd Jovan', [
...
+const BillsGrandpa3 = new NPC('Bill\'s Grandpa', [
+    'Oh, the Pokémon was called Oddish? It is adorable! Thank you so much for bringing it to me! Please, accept this Leaf Stone.',
+    'By the way, do you know of a sea Pokémon that has a red sphere on its body? You know, the one that\'s shaped like a star?',
+    'When you happen to get one, would you like to come back here so I can see it, please?',
+], {image: 'assets/images/npcs/Bill\'s Grandpa with Eevee.png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Bill\'s Grandpa Treasure Hunt', 3), new QuestLineStepCompletedRequirement('Bill\'s Grandpa Treasure Hunt', 5, GameConstants.AchievementOption.less)]),
+});
+
+const BillsGrandpa4 = new NPC('Bill\'s Grandpa', [
+    'Ah, you brought it to me. Staryu is the name, you said? It\'s certainly shaped like a star. You are so kind for coming back with it.! This Water Stone should be really useful to you.',
```
Comment (zyx-xyzzy) [inline src/scripts/towns/TownList.ts:349]: .! -> !
(in the third sentence)

File: src/scripts/towns/TownList.ts [hunk 8]
```diff
@@ -321,6 +321,70 @@ const CeruleanSuperNerd = new NPC('Super Nerd Jovan', [
...
+    'I was thinking, Bill told me about a Pokémon that is very loyal to its trainer. It\'s supposed to roar well',
```
Comment (zyx-xyzzy) [inline src/scripts/towns/TownList.ts:350]: missing punctuation at the end of the sentence, again

File: src/scripts/towns/TownList.ts [hunk 9]
```diff
@@ -321,6 +321,70 @@ const CeruleanSuperNerd = new NPC('Super Nerd Jovan', [
...
+    'I would really want to meet it, can you stop by when you meet one, please?',
```
Comment (zyx-xyzzy) [inline src/scripts/towns/TownList.ts:351]: would really want to -> either “would really like to” or “really want to”

File: src/scripts/towns/TownList.ts [hunk 10]
```diff
@@ -321,6 +321,70 @@ const CeruleanSuperNerd = new NPC('Super Nerd Jovan', [
...
+], {image: 'assets/images/npcs/Bill\'s Grandpa with Eevee.png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Bill\'s Grandpa Treasure Hunt', 5), new QuestLineStepCompletedRequirement('Bill\'s Grandpa Treasure Hunt', 7, GameConstants.AchievementOption.less)]),
+});
+
+const BillsGrandpa5 = new NPC('Bill\'s Grandpa', [
+    'Ah, so this little puppy is Growlithe? I cannot beging to imagine its roaring! I\'m so happy that you are doing all this for me. Thank you so much! I was saving this Fire Stone for a rainy day, but I want you to have it instead.',
```
Comment (zyx-xyzzy) [inline src/scripts/towns/TownList.ts:357]: beging -> begin

File: src/scripts/towns/TownList.ts [hunk 11]
```diff
@@ -321,6 +321,70 @@ const CeruleanSuperNerd = new NPC('Super Nerd Jovan', [
...
+    'Did you think I was done by the way? I wanted to ask if you knew that hugely popular Pokémon? The Pokémon that has a yellow body and red cheeks. I would love to see what it looks like.',
+    'This the one I\'m most excited to meet, so please, once you catch it come back, I will have tea prepared for you',
```
Comment (zyx-xyzzy) [inline src/scripts/towns/TownList.ts:359]: once you catch it come back -> come back once you catch it
missing punctuation at the end of the last sentence

File: src/scripts/towns/TownList.ts [hunk 12]
```diff
@@ -321,6 +321,70 @@ const CeruleanSuperNerd = new NPC('Super Nerd Jovan', [
...
+], {image: 'assets/images/npcs/Bill\'s Grandpa with Eevee.png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Bill\'s Grandpa Treasure Hunt', 7), new QuestLineStepCompletedRequirement('Bill\'s Grandpa Treasure Hunt', 9, GameConstants.AchievementOption.less)]),
+});
+
+const BillsGrandpa6 = new NPC('Bill\'s Grandpa', [
+    'Yes! That\'s the one! It\'s Pikachu, right? Ohhh, I\'ve wanted to meet it for so long. Such a cutie. You deserve this Thunder Stone!',
+    'You\'ve shown me everything I wanted to see and more. You have been a very nice trainer! My grandson could learn one thing or two from you, ho, ho!',
+    'There is one more thing I would like to ask you, would you battle this old man? I promise I\'ll make it worth your while. Let\'s go outside.',
+    'Let me tell you though, in my youth I was strong enough to beat Viridian City\'s Gym, so don\'t expect me to be a pushover',
```
Comment (zyx-xyzzy) [inline src/scripts/towns/TownList.ts:368]: one last “missing punctuation at the end of the sentence”
(btw, apologies to discord readers if all these notes clog the update channel, orz)

**Comment by LuchoDreamer** [comment]
Thank you zyx-xyzzy for going through all the dialogue and telling me the mistakes. I should have fixed them all, even one that you missed, and the punctuation during quests that you were unsure if it was intentional (it was not).

Thanks again!

File: src/scripts/quests/QuestLineHelper.ts [hunk 13]
```diff
@@ -135,6 +135,154 @@ class QuestLineHelper {
...
+            player.gainItem('Leaf_stone', 1);
+            Notifier.notify({
+                title: BillsGrandpaQuestLine.name,
+                message: 'Bill\'s Grandpa has given you a Leaf Stone.',
+                type: NotificationConstants.NotificationOption.success,
+                sound: NotificationConstants.NotificationSound.Quests.quest_ready_to_complete,
+            });
+        };
+
+        const talkToBillsGrandpa3 = new TalkToNPCQuest(BillsGrandpa3, 'Show your Oddish back to Bill\'s Grandpa.', LeafStoneReward);
```
Comment (Farboo171) [inline src/scripts/quests/QuestLineHelper.ts:190]: Same phrasing comment as Jigglypuff

File: src/scripts/towns/TownList.ts [hunk 14]
```diff
@@ -321,6 +321,70 @@ const CeruleanSuperNerd = new NPC('Super Nerd Jovan', [
...
+    'Ah, my grandson mentioned a round, green Pokémon, wait not green, it\'s blue, a blue Pokémon that has leaves growing on its head.',
+    'If you encounter that Pokémon and catch it, would you bring it here, please?',
+], {image: 'assets/images/npcs/Bill\'s Grandpa with Eevee.png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Bill\'s Grandpa Treasure Hunt', 1), new QuestLineStepCompletedRequirement('Bill\'s Grandpa Treasure Hunt', 3, GameConstants.AchievementOption.less)]),
+});
+
+const BillsGrandpa3 = new NPC('Bill\'s Grandpa', [
+    'Oh, the Pokémon was called Oddish? It is adorable! Thank you so much for bringing it to me! Please, accept this Leaf Stone.',
+    'By the way, do you know of a sea Pokémon that has a red sphere on its body? You know, the one that\'s shaped like a star?',
+    'When you happen to get one, would you like to come back here so I can see it, please?',
```
Comment (Farboo171) [inline src/scripts/towns/TownList.ts:343]: Awkward phrasing. Maybe something like "When you get one, could you come back here so I can see it, please?" for the last line

File: src/scripts/quests/QuestLineHelper.ts [hunk 15]
```diff
@@ -135,6 +135,154 @@ class QuestLineHelper {
...
+            player.gainItem('Water_stone', 1);
+            Notifier.notify({
+                title: BillsGrandpaQuestLine.name,
+                message: 'Bill\'s Grandpa has given you a Water Stone.',
+                type: NotificationConstants.NotificationOption.success,
+                sound: NotificationConstants.NotificationSound.Quests.quest_ready_to_complete,
+            });
+        };
+
+        const talkToBillsGrandpa4 = new TalkToNPCQuest(BillsGrandpa4, 'Show your Staryu back to Bill\'s Grandpa.', WaterStoneReward);
```
Comment (Farboo171) [inline src/scripts/quests/QuestLineHelper.ts:214]: Same comment as Jigglypuff step

File: src/scripts/towns/TownList.ts [hunk 16]
```diff
@@ -321,6 +321,70 @@ const CeruleanSuperNerd = new NPC('Super Nerd Jovan', [
...
+], {image: 'assets/images/npcs/Bill\'s Grandpa with Eevee.png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Bill\'s Grandpa Treasure Hunt', 3), new QuestLineStepCompletedRequirement('Bill\'s Grandpa Treasure Hunt', 5, GameConstants.AchievementOption.less)]),
+});
+
+const BillsGrandpa4 = new NPC('Bill\'s Grandpa', [
+    'Ah, you brought it to me. Staryu is the name, you said? It\'s certainly shaped like a star. You are so kind for coming back with it! This Water Stone should be really useful to you.',
+    'I was thinking, Bill told me about a Pokémon that is very loyal to its trainer. It\'s supposed to roar well.',
+    'I would really like to meet it, can you stop by when you meet one, please?',
```
Comment (Farboo171) [inline src/scripts/towns/TownList.ts:351]: Awkward phrasing. Maybe something like "I would really like to meet one, can you stop by if you capture one?" for the last line

File: src/scripts/quests/QuestLineHelper.ts [hunk 17]
```diff
@@ -135,6 +135,154 @@ class QuestLineHelper {
...
+            player.gainItem('Fire_stone', 1);
+            Notifier.notify({
+                title: BillsGrandpaQuestLine.name,
+                message: 'Bill\'s Grandpa has given you a Fire Stone.',
+                type: NotificationConstants.NotificationOption.success,
+                sound: NotificationConstants.NotificationSound.Quests.quest_ready_to_complete,
+            });
+        };
+
+        const talkToBillsGrandpa5 = new TalkToNPCQuest(BillsGrandpa5, 'Show your Growlithe back to Bill\'s Grandpa.', WaterStoneReward);
```
Comment (Farboo171) [inline src/scripts/quests/QuestLineHelper.ts:238]: Same comment as Jigglypuff

File: src/scripts/towns/TownList.ts [hunk 18]
```diff
@@ -321,6 +321,70 @@ const CeruleanSuperNerd = new NPC('Super Nerd Jovan', [
...
+], {image: 'assets/images/npcs/Bill\'s Grandpa with Eevee.png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Bill\'s Grandpa Treasure Hunt', 5), new QuestLineStepCompletedRequirement('Bill\'s Grandpa Treasure Hunt', 7, GameConstants.AchievementOption.less)]),
+});
+
+const BillsGrandpa5 = new NPC('Bill\'s Grandpa', [
+    'Ah, so this little puppy is Growlithe? I cannot begin to imagine its roaring! I\'m so happy that you are doing all this for me. Thank you so much! I was saving this Fire Stone for a rainy day, but I want you to have it instead.',
+    'Did you think I was done by the way? I wanted to ask if you knew that hugely popular Pokémon? The Pokémon that has a yellow body and red cheeks. I would love to see what it looks like.',
+    'This the one I\'m most excited to meet, so please, come back once you catch it, I will have tea prepared for you.',
```
Comment (Farboo171) [inline src/scripts/towns/TownList.ts:359]: "This the one I\'m most excited to meet Please come back once you catch it, I will have tea prepared for you.'
Lots of commas

File: src/scripts/quests/QuestLineHelper.ts [hunk 19]
```diff
@@ -135,6 +135,154 @@ class QuestLineHelper {
...
+            player.gainItem('Thunder_stone', 1);
+            Notifier.notify({
+                title: BillsGrandpaQuestLine.name,
+                message: 'Bill\'s Grandpa has given you a Thunder Stone.',
+                type: NotificationConstants.NotificationOption.success,
+                sound: NotificationConstants.NotificationSound.Quests.quest_ready_to_complete,
+            });
+        };
+
+        const talkToBillsGrandpa6 = new TalkToNPCQuest(BillsGrandpa6, 'Show your Pikachu back to Bill\'s Grandpa.', ThunderStoneReward);
```
Comment (Farboo171) [inline src/scripts/quests/QuestLineHelper.ts:262]: Same comment as Jigglypuff step

File: src/scripts/towns/TownList.ts [hunk 20]
```diff
@@ -321,6 +321,70 @@ const CeruleanSuperNerd = new NPC('Super Nerd Jovan', [
...
+    'You\'ve shown me everything I wanted to see and more. You have been a very nice trainer! My grandson could learn one thing or two from you, ho, ho!',
+    'There is one more thing I would like to ask you, would you battle this old man? I promise I\'ll make it worth your while. Let\'s go outside.',
+    'Let me tell you though, in my youth I was strong enough to beat Viridian City\'s Gym, so don\'t expect me to be a pushover.',
+], {image: 'assets/images/npcs/Bill\'s Grandpa with Eevee.png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Bill\'s Grandpa Treasure Hunt', 9), new QuestLineStepCompletedRequirement('Bill\'s Grandpa Treasure Hunt', 11, GameConstants.AchievementOption.less)]),
+});
+
+const BillsGrandpa7 = new NPC('Bill\'s Grandpa', [
+    'As I said, that was one of the best battles I\'ve ever had.',
+    'There is someone I would like to go with you, my partner Eevee.',
```
Comment (Farboo171) [inline src/scripts/towns/TownList.ts:375]: "There is someone I would like to go with you: my partner Eevee."

**Review by Farboo171 — COMMENTED** [review]
Just some text change comments

**Comment by LuchoDreamer** [comment]
Thanks for correcting the phrasing! In some places I was struggling trying to get a different wording for every time the old man asks you for a different poke. So some of them got a little bit too convoluted and unnecessary.

**Comment by Vodovik** [comment]
tested it, works well : )

File: src/scripts/temporaryBattle/TemporaryBattleList.ts [hunk 21]
```diff
@@ -223,6 +223,18 @@ TemporaryBattleList['Cue Ball Paxton'] = new TemporaryBattle(
         new TemporaryBattleRequirement('Biker Goon 3'),
     ]
 );
+TemporaryBattleList['Bill\'s Grandpa'] = new TemporaryBattle(
+    'Bill\'s Grandpa',
+    [
+        new GymPokemon('Vaporeon', 170000, 48),
+        new GymPokemon('Jolteon', 175000, 49),
+        new GymPokemon('Flareon', 180000, 50),
+    ],
```
Comment (Aegyo) [inline src/scripts/temporaryBattle/TemporaryBattleList.ts:232]: Is this intentionally ~50% tougher than the champion fight?

**Comment by LuchoDreamer** [comment]
Is it 50% tougher than Champion? I gave it 5% more hp than Giovanni. The battle is 60 seconds, not 30 seconds.

**Comment by Aegyo** [comment]
I always forget that temp battles are longer, my bad

**Review by Aegyo — APPROVED** [review]

File: src/scripts/Update.ts [hunk 22]
```diff
@@ -1933,6 +1933,9 @@ class Update implements Saveable {
             //Hex Maniac Aster
             saveData.statistics.temporaryBattleDefeated = Update.moveIndex(saveData.statistics.temporaryBattleDefeated, 112);
 
+            //Bill's Grandpa
+            saveData.statistics.temporaryBattleDefeated = Update.moveIndex(saveData.statistics.temporaryBattleDefeated, 12);
```
Comment (Aegyo) [inline src/scripts/Update.ts:1937]: Needs to be moved into a 0.10.10 update functions

**Review by Aegyo — CHANGES_REQUESTED** [review]
Just the update function needs to be moved, everything else looks fine :+1:

**Comment by LuchoDreamer** [comment]
Perfect, it's done.

**Review by Aegyo — APPROVED** [review]

**Comment by Darckclow** [comment]
Hi,
The quest is bugged. I'm at the part where I have to capture an Oddish and show it to Bill. I did it, and it validated the quest, but Bill keeps asking me to bring him an Oddish. I'm stuck in the storyline because I need to complete this quest chain to be able to battle the final gym

**Comment by DraKay** [comment]
First you should open a bug thread instead of commenting in the pr

Second did you caught all the grass pokemon and caught a oddish **again**

Third this quest is not necessary and don't lock any gym

**Pull request merged**